replay/metrics/base_metric.py

```diff
@@ -124,16 +123,17 @@ def get_enriched_recommendations(
     # if there are duplicates in recommendations,
     # we will leave fewer than k recommendations after sort_udf
     recommendations = get_top_k_recs(recommendations, k=max_k)
-    sort_udf = sf.udf(
-        sorter,
-        returnType=st.ArrayType(recommendations.schema["item_idx"].dataType),
-    )
 
     true_items_by_users = preprocess_gt(ground_truth, ground_truth_users)
     joined = (
-        recommendations.groupby("user_idx")
+        recommendations.withColumn("_num", sf.row_number().over(
```

Comment:
is duplicated along with two places in unexpectedness.py. Unify, pls, by moving this part into a separate utility function in base_metric.py.